### PR TITLE
Add detail to multiple rename deps

### LINKED ISSUE #6601: "Multiple dependencies listed for the same crate must all have the same name" error doesn't print the name of the crate
**Problem**

"Multiple dependencies listed for the same crate must all have the same name" error message contains the name of the problematic crate, but it doesn't print the name of the crate where this problem occurs. In workspaces with multiple packages it isn't so obvious where the problem is.

The problem can easily occur when giving different names in `dependencies` and `dev-dependencies` sections.

**Steps**

1. Create a new empty project.
2. Add a duplicated dependency, for example:
```toml
[dependencies]
r = { package = "rand", version = "0.6" }
ra = { package = "rand", version = "0.6" }

```
3. Try to build it.

**Notes**

Output of `cargo version`: cargo 1.32.0 (8610973aa 2019-01-02)

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #6601

### CODE REVIEW

File: src/cargo/core/resolver/resolve.rs [hunk 1]
```diff
@@ -240,14 +240,12 @@ unable to verify that `{0}` is the same as when the lockfile was generated
...
         for n in names {
-            if n == name {
-                continue;
-            }
-            failure::bail!(
-                "multiple dependencies listed for the same crate must \
-                 all have the same name, but the dependency on `{}` \
-                 is listed as having different names",
-                to
+            use failure::bail; // WORKAROUND https://github.com/rust-lang-nursery/failure/issues/299
```
[the remark this answers is not in the document]
Comment (dwijnand) [reply src/cargo/core/resolver/resolve.rs]: Thank you, Eric.  I've closed the upstream issue too.

File: src/cargo/core/resolver/resolve.rs [hunk 2]
```diff
@@ -240,14 +240,11 @@ unable to verify that `{0}` is the same as when the lockfile was generated
...
-                continue;
-            }
-            failure::bail!(
-                "multiple dependencies listed for the same crate must \
-                 all have the same name, but the dependency on `{}` \
-                 is listed as having different names",
-                to
+            failure::ensure!(
+                n == name,
+                "The crate `{}` depends on crate `{}` multiple times with different names",
```
Comment (alexcrichton) [inline src/cargo/core/resolver/resolve.rs]: as a slight nitpick we tend to prefer that error messages start with lowercase letters like the compiler

Comment (dwijnand) [reply src/cargo/core/resolver/resolve.rs]: Good to know!

**Comment by dwijnand** [comment]
Is this ready, for anyone?

**Pull request merged**